### ISSUE TITLE
questions about ValidatorDef.validateValidator

### ISSUE THREAD

**Issue by zepumph** [phetsims/axon#224]
While working on https://github.com/phetsims/axon/issues/204, I had a few questions about implementation.

* Why does this return a value? At the very least this should be documented. I'm a little unclear about where this. It seems like we are on trying to do two different things here. Return a boolean on whether or not it is valid, AND assert out if isn't valid. To me those don't mix very well.
* I see three different ways of checking if a key exists in the validator:
    * for `valueType` we just immediately assert out. As far as I know, this is an optional field. Should we have a conditional?
    * `isValidValue` uses `validator.hasOwnProperty`. That seems nice
    * `validValues` uses an `if` statement that manually checks equality to `null` or `undefined`
* Should `DEFAULT_OPTIONS` be renamed? perhaps to `DEFAULT_KEYS`?
@samreid can you please explain?

**Comment by samreid** [comment]
> * Why does this return a value?

It looks like it shouldn't return a value.  Probably vestigial.  I can't see any sites using the return value, but I can't say I understand why there is a `!` in this unit test:

```js
window.assert && assert.throws( () => !ValidatorDef.validateValidator( { valueType: Array, isValidValue: 4 } ), 'isValidValue should be function' );
```

> * I see three different ways of checking if a key exists in the validator:

Checking for `hasOwnProperty` for each case sounds great to me.

> * Should `DEFAULT_OPTIONS` be renamed? perhaps to `DEFAULT_KEYS`?

Maybe this doesn't need to be mixed into Property?  Perhaps instead Property should just document "adheres to the Validator specification", but without attempting to do that by copying keys?

**Comment by zepumph** [comment]
most of this was fixed as part of work done today in #204.

All that is left is to use `hasOwnProperty` instead of checking that they are truthy.

**Comment by zepumph** [comment]
Fixed above. Closing
